loaders/src/glTF/2.0/glTFLoader.ts

```diff
@@ -186,7 +186,7 @@ export class GLTFLoader implements IGLTFLoader {
      * The root Babylon mesh when loading the asset.
      */
     public get rootBabylonMesh(): Mesh {
-        return this._rootBabylonMesh;
+        return this._rootBabylonMesh!;
```

Comment:
Note that `!` here is continuing the existing lie that this cannot be undefined, while the bug showed that it can be! I am not happy about this, but changing it would affect other pieces of code (e.g. extensions), and all TypeScript users that use this directly.

Happy to change that (to be nullable) if you agree!?

Note that the same "problem" applies to many other properties here and probably also in many other places, that are optimistically typed. A `strict` (or at least `strictPropertyInitialization: true`) TS config would catch all these cases (by complaining about `Property '...' has no initializer and is not definitely assigned in the constructor`), but this would probably affect most parts of the whole code base...

Reply:
Yeah, I wish we would turn this on, but it will take some effort.

Reply:
> Happy to change that (to be nullable) if you agree!?

I agree but it is unfortunately a breaking change. Maybe you can throw an exception instead?

Reply:
Hm, does that make it less of a breaking change?

> it is unfortunately a breaking change

Yes, for TS-consumers at least, but wouldn't that be ok, as it will only be released as part of the next major, right?

Reply:
I would suggest to make it nullable and reference it in our what s new breaking change list. The impact is mostly internal as it does not seem to be used a lot outside; The playground does show no usage at all for instance: https://doc.babylonjs.com/playground?q=rootBabylonMesh&type=code

Reply:
@simonihmig, thinking about it more, lets make it nullable for sure :-)

Reply:
@sebavan done! Had to update smaller parts of some glTF extensions. Although I don't know the code of those, the changes *should* be syntactically correct (i.e. don't change behavior), but please double check!